src/exchange_adapters/alphavantage.ts

```diff
@@ -0,0 +1,142 @@
+   *    }
+   *  }
+   */
+  parseTicker(json: Response): Ticker {
+    const response = json['Realtime Currency Exchange Rate']
+
+    const from = response[ResponseKeys.fromCurrency]
+    const to = response[ResponseKeys.toCurrency]
+    assert(
+      from === this.config.baseCurrency,
+      `From currency mismatch in response: ${from} != ${this.config.baseCurrency}`
+    )
+    assert(
+      to === this.config.quoteCurrency,
+      `To currency mismatch in response: ${to} != ${this.config.quoteCurrency}`
+    )
+
+    const ticker = {
+      ...this.priceObjectMetadata,
+      ask: this.safeBigNumberParse(response[ResponseKeys.ask])!,
+      bid: this.safeBigNumberParse(response[ResponseKeys.bid])!,
+      lastPrice: this.safeBigNumberParse(response[ResponseKeys.rate])!,
+      timestamp: this.toUnixTimestamp(response[ResponseKeys.lastUpdated]),
+      // These FX API's do not provide volume data,
+      // therefore we set all of them to 1 to weight them equally
+      baseVolume: new BigNumber(1),
+      quoteVolume: new BigNumber(1),
+    }
+    this.verifyTicker(ticker)
+    return ticker
+  }
+
+  toUnixTimestamp(datetime: string): number {
+    // datetime format: YYYY-MM-DD HH:MM:SS
+    const date = datetime.split(' ')[0]
+    const time = datetime.split(' ')[1]
+
+    const [year, month, day] = date.split('-').map(Number)
+    const [hours, minutes, seconds] = time.split(':').map(Number)
+
+    // month should be 0-indexed
+    return Date.UTC(year, month - 1, day, hours, minutes, seconds) / 1000
+  }
+
+  async isOrderbookLive(): Promise<boolean> {
+    /*
+      This check requires us to make a call to the ticker endpoint, however we can't
+      use fetchTicker() because it creates a circular dependency: 
+      fetchTicker requires isOrderbookLive under the hood, which requires fetchTicker, etc.
+
+      Therefore we make a similar call using fetchFromApi() and the orderbook status type.
+
+      We will have the same scenario for other adapters as well so we can find a nicer 
+      way to do this in the future. 
+    */
+    const base = this.config.baseCurrency
+    const quote = this.config.quoteCurrency
+
+    const tickerJson: Response = await this.fetchFromApi(
+      ExchangeDataType.ORDERBOOK_STATUS,
+      `query?function=CURRENCY_EXCHANGE_RATE&from_currency=${base}&to_currency=${quote}&apikey=${this.config.apiKey}`
+    )
+    const ticker = this.parseTicker(tickerJson)
+
+    /*
+      TODO: verify the behaviour of the ticker endpoit on the weekend.
+
+      The check below assumes that the timestamp returned will stale on
+      the weekend while the markets are closed.
+    */
+    const thirtyMinutesInSecs = 30 * 60
+
+    const now = Date.now() / 1000
+    const lastUpdated = ticker.timestamp // timestamp is already in seconds
+
+    assert(now >= lastUpdated, `Ticker timestamp is in the future: ${now} < ${lastUpdated}`)
+    return now - lastUpdated <= thirtyMinutesInSecs
```

Comment:
If I understand correctly they don't actually return anything to indicate that the order book is closed, so we will manually pick a weekend range where we consider the order book not to be live.

Reply:
the approach we will take is to find another source that returns stale data on the weekends and make the configuration n/n for the oracles to report. But yes, in this case as this exchange doesn't return anything I removed all of this code and just made this method return `true`